### PR TITLE
FEATURE: Add relation between `TopicChatChannel` and `Chat::Channel`

### PR DESCRIPTION
**Pull request by Grubba27**
- Add tests for `TopicChatChannel` and `Chat::Channel` relation
- Add tests for `handle_topic_chat_channel_creation`

### CODE REVIEW

File: lib/discourse_livestream/chat_channel_extension.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+module DiscourseLivestream
+  module ChatChannelExtension
+    extend ActiveSupport::Concern
+
+    prepended do
+      has_one :topic_chat_channel,
```
Comment (markvanlan) [inline lib/discourse_livestream/chat_channel_extension.rb:8]: this is a bit generic, could it be `livestream_topic_chat_channel`?

Comment (Grubba27) [reply lib/discourse_livestream/chat_channel_extension.rb:8]: let me test, I copied from what we have from topic: https://github.com/discourse/discourse-livestream/blob/eb5eab9869d6dd807a3d2e3dd856c2d3757c48a6/lib/discourse_livestream/topic_extension.rb#L8-L9

Comment (Grubba27) [reply lib/discourse_livestream/chat_channel_extension.rb:8]: using this way gives me this error:
Screenshot 2024-08-16 at 15 53 11

Comment (markvanlan) [reply lib/discourse_livestream/chat_channel_extension.rb:8]: did you remove the `class_name` key? Keep that there as you had it

**Review by jmperez127 — APPROVED** [review]
LGTM, but I do agree with @markvanlan. If you can change the naming it would be great

**Pull request merged**